azure-kusto-data/azure/kusto/data/_kusto_client.py

```diff
@@ -223,6 +202,135 @@ def _get_table(self, table_id):
+        for i in range(0, len(query_status_table)):
+            if query_status_table[i][self._error_column] < 4:
+                result.append(
+                    "Please provide the following data ot Kusto: CRID='{0}' Description:'{1}'".format(
+                        query_status_table[i][self._crid_column],
+                        query_status_table[i][self._status_column],
+                    )
+                )
+        return result
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._index >= self.tables_count:
+            raise StopIteration
+        result = self.tables[self._index]
+        self._index += 1
+        return result
+
+    def __getitem__(self, key):
+        if isinstance(key, numbers.Number):
+            return self.tables[key]
+        return self.tables[self._tables_names.index(key)]
+
+    def __len__(self):
+        return self.tables_count
+
+    def to_dataframe(self):
+        return pandas.DataFrame(data=self.tables, columns=[t.table_name for t in self.tables])
+
+
+class _KustoResponseDataSetV1(_KustoResponseDataSet):
+
+    _status_column = "StatusDescription"
+    _crid_column = "ClientActivityId"
+    _error_column = "Severity"
+    _v1_tables_kinds = {
+        "QueryResult": WellKnownDataSet.PrimaryResult,
+        "@ExtendedProperties": WellKnownDataSet.QueryProperties,
+        "QueryStatus": WellKnownDataSet.QueryCompletionInformation,
+    }
+
+    def __init__(self, json_response):
+        if len(json_response["Tables"]) <= 2:
+            json_response["Tables"][0]["TableKind"] = WellKnownDataSet.PrimaryResult
+            json_response["Tables"][0]["TableId"] = 0
+
+            if len(json_response["Tables"]) == 2:
+                json_response["Tables"][1]["TableKind"] = WellKnownDataSet.QueryProperties
+                json_response["Tables"][1]["TableId"] = 1
+            self.tables = [_KustoResultTable(t) for t in json_response["Tables"]]
+            self.tables_count = len(json_response["Tables"])
+        else:
+            json_tables = json_response["Tables"][:-1]
+            toc = _KustoResultTable(json_response["Tables"][-1])
+            for i in range(0, len(json_tables)):
+                json_table = json_tables[i]
+                json_table["TableName"] = toc[i]["Name"]
+                json_table["TableId"] = toc[i]["Id"]
+                json_table["TableKind"] = self._v1_tables_kinds[toc[i]["Kind"]]
+                self.tables.append(_KustoResultTable(json_table))
+
+        super(_KustoResponseDataSetV1, self).__init__()
+
+
+class _KustoResponseDataSetV2(_KustoResponseDataSet):
+
+    _status_column = "Payload"
+    _error_column = "Level"
+    _crid_column = "ClientRequestId"
+
+    def __init__(self, json_response):
+        json_response = [t for t in json_response if t["FrameType"] == "DataTable"]
+        for json_table in json_response:
+            json_table["TableKind"] = WellKnownDataSet[json_table["TableKind"]]
```

Comment:
Intentionally modifying the contents of an argument?

Reply:
Yes. I want the table kind to be from the defined enum. The response us just text.